ahds/header.py

```diff
@@ -8,27 +8,102 @@
 ::
 
     >>> from amira.header import AmiraHeader
-    >>> ah = AmiraHeader.from_file('file.am')
+    >>> ah = AmiraHeader.from_file('<somfile>.am')
     >>> print ah
 
 Each nested object is constructed from the :py:class:``Block`` class defined.
 
-There are four top-level attributes that every ``AmiraHeader`` will have:
+There are fife top-level attributes that every ``AmiraHeader`` will have:
 
 *    designation
 
 *    definitions
 
 *    parameters
 
+*    materials
+
 *    data_pointers
 
-Each attribute can be queried using the ``attrs`` attribute.
+All of them, exempt materials and parameters are marked deprecated and may be removed in
+future. The designation and defintions attributes just return the reference to the AmiraHeader
+instance which host all the attributes and data array declarations (definitions) found within
+the AmiraMesh and HyperSurface files. The data_pointers attribute collects from each
+attribute representing a valid array declaration the defined data definitions and returns the resulting
+list. New code should directly access the Block objects describing the available data directly
+from the corresponding array attribute of the AmiraHeader object. 
+
+The blocks defining the available data structures extend the basic Block to the corresponding 
+AmiraMeshDataStream and AmiraHxSurfaceDataStream blocks which allow to load and access the correspinding
+data using the data property defined by theses blocks. The additional stream_data, encoded_data and
+decoded_data attributes are deprecated and may be removed in future versions.
+
+The attrs attribute of all Blocks and extended Block objects including AmiraHeader it self
+which allows to query the attributes added during loading is deprecated and may be removed in future version.
+To query which attributes are defined on each block use the builtin dir function instead. To 
+access the defined attributes just use them liḱe any other static attribute.
 
+The following attributes are moved from the designation attribute to the basic header block:
+        
+        *    filetype e.g. ``AmiraMesh`` or ``HyperSurface``
+        *    dimensions e.g. ``3D``
+        *    format e.g. ``BINARY-LITTLE-ENDIAN``
+        *    version e.g. ``2.1``
+        *    extra_format (if present)
+        
 :: 
 
-    >>> print ah.data_pointers.attrs
-    ['data_pointer_1', 'data_pointer_2', 'data_pointer_3', 'data_pointer_4', 'data_pointer_5', 'data_pointer_6']
+    >>> print(ah.Nodes)
+    print(ah.Nodes)
+     +-Nodes: (<ahds.ahds_common.Block object at 0x7f5dc1a74d08>)
+     |  |  +-Coordinates: (<ahds.data_stream.AmiraMeshDataStream object at 0x7f5dc1a97458>)
+     |  |  |  +-block: 1
+     |  |  |  +-array: Nodes(<ahds.ahds_common.Block object at 0x7f5dc1a74d08>)
+     |  |  |  +-type: float
+     |  |  |  +-dimension: 3
+     |  +-dimension: 3154
+    >>> print(ah.Tetrahedra.Nodes)
```

Comment:
How do we know there is a Tetrahedra block here?

Reply:
For AmiraMesh file that is implicitly done as loading in a first step just reads the header without loading any data as done before. And not before accessing ah.Tetrahedra.Nodes.Data any data is loaded. There thechnically nothing changed. Exempt that Tetrahedra is the definition of array in the top lines of Amira mehs file while Nodes is the name of data form the data descriptor line

Tetrahedra { int [4] Nodes } @1

And as AmiraMesh has changed some time the array definition (need to check pdf) i decided to allways use the new name.

Reading encoded data from file and decoding are two distinct steps. The first is happening when anyway bypassing while looking for data block of interest. Deconding happens when data attribute is accessed for the first time.

Reply:
Perhaps we can have a data folder inside the ahds folder. At least for a start. Please choose then we can refine. Thanks very much.

Reply:
Ok i will try to do forth comming weekend when work on grant proposal is done had just a one day break for simplifying code. Will put some example data three and i also will add some test.py files inside a simple-tests folder we can remove afterwards i just do not want to collide with default github structure and at the same time i do not want to litter root